### DISCUSSION TITLE
Wrong API MIME Type

### DISCUSSION THREAD

**Discussion by ibnesayeed** [cs531-f22/cs531-f22.github.io#25] [Assignment]
Find an "in the wild" examples of an API with a MIME type of `application/json` that should be something else and suggest what the MIME type should be.

**Comment by calebkbrad** [comment]
This API call (https://dog.ceo/api/breeds/image/random) is supposed to simply return a random image of a dog. However, instead of returning an image file directly, it returns a JSON object with a link to that image, even when the call is made through a browser. I would argue that the API should return an image/* MIME type directly, instead of having a layer of indirection. 

```
$ curl -i https://dog.ceo/api/breeds/image/random
HTTP/2 200
date: Thu, 01 Dec 2022 16:59:25 GMT
content-type: application/json
x-powered-by: PHP/8.1.0
cache-control: no-cache, private
access-control-allow-origin: *
via: 1.1 varnish (Varnish/6.3), 1.1 varnish (Varnish/6.3)
x-cache-hits: 0
x-cache: MISS
age: 0
vary: Accept-Encoding
cf-cache-status: DYNAMIC
report-to: {"endpoints":[{"url":"https:\/\/a.nel.cloudflare.com\/report\/v3?s=xU05Pa5D9YzunQ39q1Hi3k1ShtWmz7AEdEJfvEq8W%2B3djk49dCU%2BwM4NsoOS14YdgEOTM%2Fs9s0MW8hNhxeqfoeMjhTV7FQrMr7mxAC%2FuVGZhgYf1eaPQHX55"}],"group":"cf-nel","max_age":604800}
nel: {"success_fraction":0,"report_to":"cf-nel","max_age":604800}
server: cloudflare
cf-ray: 772d5f8dcfc95961-IAD
alt-svc: h3=":443"; ma=86400, h3-29=":443"; ma=86400

{"message":"https:\/\/images.dog.ceo\/breeds\/terrier-american\/n02093428_12590.jpg","status":"success"}
```

**Comment by ibnesayeed** [comment]
@calebkbrad in this case the `Content-Type` header is reporting the correct MIME type of the payload. The intent of the endpoint is not to be judged here, but see if the returned content is accurately described by the said header.

**Reply by calebkbrad** [reply]
Ah, okay I apologize I misunderstood.
